module/zfs/zfs_log.c

```diff
@@ -212,6 +212,31 @@ zfs_log_fuid_domains(zfs_fuid_info_t *fuidp, void *start)
 }
 
 /*
+ * If zp is an xattr node, check whether the xattr owner is unlinked.
+ * We don't want to log anything if the owner is unlinked.
+ */
+static int
+zfs_xattr_owner_unlinked(znode_t *zp)
+{
+	int unlinked = 0;
+	znode_t *dzp;
+	igrab(ZTOI(zp));
+	while (zp->z_pflags & ZFS_XATTR) {
```

Comment:
The need for this loop is subtle and should be better explained.  It wasn't at all clear to me initially that what you're doing here is walking backwards through the tree of xattrs to determine if any of the parents were unlinked.